flow/Trace.cpp

```diff
@@ -1442,3 +1475,27 @@ std::string traceableStringToString(const char* value, size_t S) {
 
 	return std::string(value, S - 1); // Exclude trailing \0 byte
 }
+
+RealTimePrinter::RealTimePrinter() : beginTime(Clock::now()), flowBeginTime(g_network->now()) {}
+
+// converts the given flow time into a
+// string. now has to be larger
+// then flowBeginTime.
+// %Y-%m-%dT%H:%M:%S
+// This only has second-resolution for the simple reason
+// that std::put_time does not support higher resolution.
+// As we always also log the flow time and this is only for
+// human readability, this is good enough for now.
+std::string RealTimePrinter::toString(double now) {
+	time_t n = Clock::to_time_t(beginTime + std::chrono::microseconds(std::lround(1e6 * (now - flowBeginTime))));
+	if (g_network->isSimulated()) {
```

Comment:
If we make the above change, we'll need to do this too:

```suggestion
	if (g_network && g_network->isSimulated()) {
```

Note that simulation doesn't support early trace events, so it doesn't matter if we use a weird time for them in simulation.

Reply:
Ah, i see.  Why not just convert 'now' to time?  The issue with calling g_network->now() here as opposed to using the argument is we won't be using the same time as was passed to the Trace event.  Admittedly its unlikely to make a difference since DateTime is at the second granularity...

Reply:
I'm not sure I understand the question exactly, but to clarify my comment:

The idea here is that we would use the real time for simulated events for the reasons you stated. However, before we have a network set up, we can't know if we are in simulation. If we use the condition I proposed above, we would assume the answer is no and then use whatever value was passed in as `now`. That may generate weird times for your simulated runs. However, we don't support logging trace events created before the network is created in simulation, so it doesn't actually matter.

Reply:
The reason we don't use 
Clock::to_time_t(<passed var>
instead of 
Clock::to_time_t(g_network->now())

is to avoid skew between the detail 'Time' and detail 'DateTime'.  If both detail objects call g_network->now(),  DateTime is not quite identical to Time.  Admittedly, since DateTime is only in 'seconds' granularity, it likely doesn't matter